### PR TITLE
refactor:adding chainId check in EvmTransactionLog

### PR DESCRIPTION
**Pull request by 3scava1i3r**
---
name: 'Better equality check for EvmTransactionLog, it should also check for chainId'
---

### Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/SECURITY.md).
- [x] My code is conform the [code style](https://github.com/MoralisWeb3/Moralis-JS-SDK/blob/main/CODE_STYLE.md)
- [x] I have made corresponding changes to the documentation
- [x] I have updated Typescript definitions when needed

### Issue Description

Related issue: N/A
### Solution Description

Added chainId as a optional parameter in EvmTxnLogInput,added the same in equals functions for equality check

### CODE REVIEW

File: .changeset/thick-pugs-live.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@moralisweb3/evm-utils': major
```
Comment (b4rtaz) [inline .changeset/thick-pugs-live.md:2]: Should be `patch`.

**Review by b4rtaz — CHANGES_REQUESTED** [review]

File: .changeset/thick-pugs-live.md (hunk 1 above)
Comment (3scava1i3r) [reply .changeset/thick-pugs-live.md]: changed the same

**Review by b4rtaz — APPROVED** [review]

**Review by ErnoW — CHANGES_REQUESTED** [review]
Please check the pipelines there are some issues:
- small formatting issues. Make sure you have prettier installed as extention in your editor. Or run `yarn format`
- The places where `EvmTransactionLog` are implemented (in the evmApi) should provide the chainId to the `EvmTransactionLog`

**Comment by 3scava1i3r** [comment]
Will do,also i found an issue with EVMTXNReceipt but was not able to find the same,do you mind pointing me to that

**Comment by ErnoW** [comment]
> EVMTXNReceipt

The EvmTransactionReceipt has been removed, we now only use EvmTransactionResponse.
(Previously we had a difference between a receipt and response, but now we have all the data in EvmTransactionResponse)

**Comment by 3scava1i3r** [comment]
@ErnoW an issue that i encountered was the evmTransactionlogish is imported from node_modules evmUtils which is not updated with `chainId` therefore it produces the error of 

`Object literal may only specify known properties, and 'chainId' does not exist in type 'EvmTransactionLogish'.`
I'd appreciate any help here

**Comment by ErnoW** [comment]
Fixed some small issues and moved to https://github.com/MoralisWeb3/Moralis-JS-SDK/pull/762 as I cannot push to this branch

**Pull request closed without merging**